src/pubsub/ua_pubsub_reader.c

```diff
@@ -1299,6 +1312,21 @@ decodeNetworkMessage(UA_Server *server, UA_ByteString *buffer, size_t *pos,
     return UA_STATUSCODE_GOOD;
 }
 
+#ifdef UA_ENABLE_JSON_ENCODING
+UA_StatusCode
+decodeNetworkMessageJson(UA_Server *server, UA_ByteString *src,
+                         UA_NetworkMessage *dst, UA_PubSubConnection *connection) {
+#ifdef UA_DEBUG_DUMP_PKGS
+    UA_dump_hex_pkg(buffer->data, buffer->length);
+#endif
+    UA_StatusCode rv = UA_NetworkMessage_decodeJson(dst,src);
+    UA_CHECK_STATUS_ERROR(rv, return rv,
```

Comment:
That function does not much besides calling UA_NetworkMessage_decodeJson.
Could be unwrapped inside of decodeAndProcessNetworkMessageJson to reduce the levels of indirection.